### DISCUSSION TITLE
Image upload option not available in the latest version

### DISCUSSION THREAD

**Discussion by indushanosos** [sibiraj-s/ngx-editor#522] [General]
![image](https://github.com/sibiraj-s/ngx-editor/assets/136599858/98ef3a4e-616b-419b-81ae-b314436aa0e7)

this is my older plugin : 
![image](https://github.com/sibiraj-s/ngx-editor/assets/136599858/adacdca2-0837-4b6b-a025-98a101d35d77)

Dear ngx-editor Team,

I hope this message finds you well. I am reaching out to report an issue I encountered while using the ngx-editor plugin version 4.1.0 in my Laravel backend environment.

The problem arises specifically with the image upload functionality. While attempting to upload JPEG (.jpg) images through the ngx-editor interface, I noticed that these images are not correctly decoded in the backend. However, PNG (.png) images upload successfully without any issues.

Upon further investigation, I discovered that the image upload feature in the newer version of ngx-editor does not seem to provide the same functionality as the older version that I was using. As a result, I am unable to address this issue by simply updating to a newer version of the plugin.

Given the importance of image upload functionality for my project, I would greatly appreciate any guidance or assistance you can provide in resolving this issue. Is there an alternative solution or workaround that I can implement to ensure successful JPEG image uploads while using the latest version of ngx-editor?

Thank you for your attention to this matter. I look forward to your prompt response and assistance in resolving this issue.

**Comment by sibiraj-s** [comment]
Hi, You'll need to create your own extension. Since each project handles image uploads differently, its not possible to cover all scenarios, hence it was removed few versions back.

This editor is now powered by prosemirror, You can build a prosemirror extension to handle this. There is also an example for the extension https://prosemirror.net/examples/upload/

You can refer here to create custom menu items https://sibiraj-s.github.io/ngx-editor/menu/#custom-menu also browse through codebase to see how the modules are built.

[a comment by an automated account is not shown]
